core/src/main/java/org/elasticsearch/index/query/TermsQueryParser.java

```diff
@@ -94,6 +98,8 @@ public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOExcept
                 } else {
                     throw new ParsingException(parser.getTokenLocation(), "[terms] query does not support [" + currentFieldName + "]");
                 }
+            } else {
+                throw new ParsingException(parser.getTokenLocation(), "[terms] unknown token [" + token + "] after [" + currentFieldName + "]");
```

Comment:
You switched to using the query builder NAME in other places - why not here as well?